parse_date.re

```diff
@@ -707,25 +707,50 @@ static const timelib_tz_lookup_table* abbr_search(const char *word, timelib_long
 	return NULL;
 }
 
+#define TIMELIB_TP_VALUE(tp) tp->gmtoffset - (tp->type * 3600)
+
 static timelib_long timelib_lookup_abbr(char **ptr, int *dst, char **tz_abbr, int *found)
 {
 	char *word;
 	char *begin = *ptr, *end;
 	timelib_long  value = 0;
 	const timelib_tz_lookup_table *tp;
 
+	int has_dash = 0;
 	while (**ptr != '\0' && **ptr != ')' && **ptr != ' ') {
-		++*ptr;
+		if(*((*ptr)++) == '-'){
+			has_dash = 1;
+		}
```

Comment:
Wouldn't it just be simpler to make sure the last character is not a `-` and then rewind with using `--*ptr` ?

Reply:
Now that i think of it, is it even possible for a timezone *abbreviation* to hold dashes? if not, this while check can be rewritten altogether

Reply:
Also what is the "last character" in this scope?
We don't know where the timezone part actually ends before doing a lookup, take for example "CET-2020-02". The last character here would be 2 with the current while() loop

Reply:
Part of the problem seems to be that this code is used for both abbreviated forms (like "CET") and non abbreviated forms (like "America/Port-au-Prince").

While i agree that using the latter is a weird format, i wouldn't say the same for the abbreviated form.
The abbreviated form has no symbol or special character (https://en.wikipedia.org/wiki/List_of_time_zone_abbreviations). This means that it could be handled separately in another function